x-pack/auditbeat/module/system/sockets/sockets_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build linux
+
+package sockets
+
+import (
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestData(t *testing.T) {
+	f := mbtest.NewReportingMetricSetV2(t, getConfig())
+	err := mbtest.WriteEventsReporterV2(f, t, "")
+	if err != nil {
+		t.Fatal("write", err)
+	}
+}
+
+func TestFetch(t *testing.T) {
+	// Consume first event: list of all currently open sockets
+	ms := mbtest.NewReportingMetricSetV2(t, getConfig())
+	events, errs := mbtest.ReportingFetchV2(ms)
+	if errs != nil {
+		t.Fatal("fetch", errs)
+	}
+	_, err := events[0].MetricSetFields["socket"].([]common.MapStr)[0].HasKey("local.port")
+	assert.NoError(t, err)
+
+	ln, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ln.Close()
+
+	addr := ln.Addr().String()
+	i := strings.LastIndex(addr, ":")
```

Comment:
This seems to assume IPv4. It's a test, so perhaps fine. But make sure you don't make this assumption in the main code :-)

Reply:
Should be fine in the main code.